infogami/utils/storage.py

```diff
@@ -206,25 +206,39 @@ class DictPile(DictMixin):
     """
     def __init__(self, dicts=[]):
         self.dicts = dicts[:]
-        
+
+    def __len__(self):
+        return len(self.dicts)
```

Comment:
I think len(self.keys()) is a more logical implementation.

Reply:
Why is it more logical?

Reply:
All the standard methods work on the merged entries from all the dicts. The only dict-specific method is add_dict. This currently returns the number of dicts, not the number of visible merged entries in the union of the dicts, as I'd expect.

Reply:
done, thanks for the review @tfmorris